presto-main/src/test/java/com/facebook/presto/operator/aggregation/AggregationTestUtils.java

```diff
@@ -89,27 +108,27 @@ public static Block getFinalBlock(GroupedAccumulator accumulator)
         return blockBuilder.build();
     }
 
-    private static void assertAggregationInternal(InternalAggregationFunction function, Object expectedValue, Page... pages)
+    private static void assertAggregationInternal(InternalAggregationFunction function, BiFunction<Object, Object, Boolean> isEqual, String testDescription, Object expectedValue, Page... pages)
     {
-        BiConsumer<Object, Object> equalAssertion = (actual, expected) -> {
-            assertEquals(actual, expected);
-        };
-        if (expectedValue instanceof Double && !expectedValue.equals(Double.NaN)) {
-            equalAssertion = (actual, expected) -> assertEquals((double) actual, (double) expected, 1e-10);
-        }
-        if (expectedValue instanceof Float && !expectedValue.equals(Float.NaN)) {
-            equalAssertion = (actual, expected) -> assertEquals((float) actual, (float) expected, 1e-10f);
-        }
-
         // This assertAggregation does not try to split up the page to test the correctness of combine function.
         // Do not use this directly. Always use the other assertAggregation.
-        equalAssertion.accept(aggregation(function, pages), expectedValue);
-        equalAssertion.accept(partialAggregation(function, pages), expectedValue);
+        assertFunctionEquals(isEqual, testDescription, aggregation(function, pages), expectedValue);
+        assertFunctionEquals(isEqual, testDescription, partialAggregation(function, pages), expectedValue);
         if (pages.length > 0) {
-            equalAssertion.accept(groupedAggregation(function, pages), expectedValue);
-            equalAssertion.accept(groupedPartialAggregation(function, pages), expectedValue);
-            equalAssertion.accept(distinctAggregation(function, pages), expectedValue);
+            assertFunctionEquals(isEqual, testDescription, groupedAggregation(function, pages), expectedValue);
+            assertFunctionEquals(isEqual, testDescription, groupedPartialAggregation(function, pages), expectedValue);
+            assertFunctionEquals(isEqual, testDescription, distinctAggregation(function, pages), expectedValue);
```

Comment:
`groupedPartialAggregation` and `distinctAggregation` appear to use `Assert.assertEquals` to check aggregation results. Is this the case? If so, we need to pass a different version of `Assert.assertEquals`.